pkg/server/commands/listusers/list_users_rpc.go

```diff
@@ -69,14 +69,20 @@ func (l *listUsersQuery) ListUsers(
 	if !ok {
 		return nil, fmt.Errorf("typesystem missing in context")
 	}
-	hasPossibleEdges, err := doesHavePossibleEdges(typesys, req)
-	if err != nil {
-		return nil, err
-	}
-	if !hasPossibleEdges {
-		return &openfgav1.ListUsersResponse{
-			Users: []*openfgav1.User{},
-		}, nil
+
+	userFilter := req.GetUserFilters()[0]
+	isReflexiveUserset := userFilter.GetType() == req.GetObject().GetType() && userFilter.GetRelation() == req.GetRelation()
```

Comment:
@jon-whit @miparnisari As we discussed, this check for reflexive usersets has now been separated from `doesHavePossibleEdges`.